osf/models/user.py

```diff
@@ -456,10 +457,17 @@ def social_links(self):
         social_user_fields = {}
         for key, val in self.social.items():
             if val and key in self.SOCIAL_FIELDS:
-                if not isinstance(val, basestring):
-                    social_user_fields[key] = val
+                if isinstance(self.SOCIAL_FIELDS[key], basestring):
+                    if isinstance(val, basestring):
+                        social_user_fields[key] = self.SOCIAL_FIELDS[key].format(val)
+                    else:
+                        # Provide a single account url for services which we allow multiple-accounts
+                        social_user_fields[key] = self.SOCIAL_FIELDS[key].format(val[0])
                 else:
-                    social_user_fields[key] = self.SOCIAL_FIELDS[key].format(val)
+                    if not isinstance(val, basestring):
+                        social_user_fields[key] = val
+                    else:
+                        social_user_fields[key] = [val]
         return social_user_fields
```

Comment:
I'm thinking we can improve readability a bit, it takes a minute to break down what is happening.  To start, I'd recommend parallel structure, in the first if, you're checking if it *is* a basestring, in the second conditional, you're checking if it's *not* a basestring. I'd make them the same.

Reply:
I attempted to address this with the parallel structure and also added a doc string to clarify the purpose of the function.